tests/test_contact_wrapper_classes.py

```diff
@@ -144,121 +99,91 @@ def test_external_contact_rod_rigid_body_without_collision():
+    ext_contact = ExternalContact(k=1.0, nu=0.0)
     mock_rod_one_external_forces_before_execution = mock_rod_one.external_forces.copy()
     mock_rod_two_external_forces_before_execution = mock_rod_two.external_forces.copy()
     ext_contact.apply_forces(mock_rod_one, 0, mock_rod_two, 0)
 
     assert_allclose(
-        mock_rod_one.external_forces,
-        mock_rod_one_external_forces_before_execution,
-        rtol=tol,
-        atol=tol,
+        mock_rod_one.external_forces, mock_rod_one_external_forces_before_execution
     )
-
     assert_allclose(
-        mock_rod_two.external_forces,
-        mock_rod_two_external_forces_before_execution,
-        rtol=tol,
-        atol=tol,
+        mock_rod_two.external_forces, mock_rod_two_external_forces_before_execution
     )
 
 
 def test_self_contact_with_rod_self_collision():
 
     "Testing Self Contact wrapper rod self collision with analytical verified values"
 
-    tol = 1e-6
     mock_rod = MockRod()
-    sel_contact = SelfContact(k=1.0, nu=0.5)
+
+    "Test values have been copied from 'test_contact_specific_functions.py/test_calculate_contact_forces_self_rod()'"
+    mock_rod.n_elems = 4
+    mock_rod.position_collection = np.array([[1, 3, 5, 7], [0, 0, 0, 0], [0, 0, 0, 0]])
+    mock_rod.radius = np.array([1, 1, 1, 1])
+    mock_rod.lengths = np.array([3.0, 3.0, 3.0, 3.0])
+    mock_rod.tangents = np.array(
+        [[1.0, 1.0, 1.0, 1.0], [0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0]]
+    )
+    mock_rod.velocity_collection = np.array(
+        [[0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0]]
+    )
+    mock_rod.internal_forces = np.array(
+        [[0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0]]
+    )
+    mock_rod.external_forces = np.array(
+        [[0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0]]
+    )
+    sel_contact = SelfContact(k=1.0, nu=0.0)
     sel_contact.apply_forces(mock_rod, 0, mock_rod, 0)
 
     assert_allclose(
         mock_rod.external_forces,
-        np.array(
-            [
-                [-0.976629, -0.663611, -0.934193, -0.471303, -0.028112],
-                [-1.125742, -2.077304, -1.465054, -0.595825, -0.876467],
-                [0.204132, 0.967667, 0.062506, -0.299531, 0.776233],
-            ]
-        ),
-        rtol=tol,
-        atol=tol,
+        np.array([[-0.33, -0.66, 0.5, 0.5], [0, 0, 0, 0], [0, 0, 0, 0]]),
+        atol=1e-2,
     )
 
 
 def test_self_contact_with_rod_no_self_collision():
 
     "Testing Self Contact wrapper rod no self collision with analytical verified values"
 
-    tol = 1e-6
     mock_rod = MockRod()
-    mock_rod.position_collection = np.array(
-        [[1, 3, 5, 7, 9], [11, 13, 15, 17, 19], [21, 23, 25, 27, 29]]
-    )
-    sel_contact = SelfContact(k=1.0, nu=0.5)
+
+    "setting rod elements position such that tehre is no self collision"
+    mock_rod.position_collection = np.array([[1, 5], [0, 0], [0, 0]])
```

Comment:
if you change the rod position_collection make sure that you also update the other properties with it too. The original rod had 4 elements now your updated rod has 1 element so all the other properties should be updated to have 1 element too. Alternatively you can make this rod have the same number of elements as before, just make sure all the properties still make sense after this update. Also, typo there*